OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitYoutubeVideoFragment.java

```diff
@@ -0,0 +1,405 @@
+package org.edx.mobile.view;
+
+import android.content.pm.ActivityInfo;
+import android.content.res.Configuration;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Handler;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.Toast;
+
+import com.google.android.youtube.player.YouTubeInitializationResult;
+import com.google.android.youtube.player.YouTubePlayer;
+import com.google.android.youtube.player.YouTubePlayer.Provider;
+import com.google.android.youtube.player.YouTubePlayerSupportFragment;
+
+import org.edx.mobile.R;
+import org.edx.mobile.base.BaseFragmentActivity;
+import org.edx.mobile.model.api.TranscriptModel;
+import org.edx.mobile.model.course.VideoBlockModel;
+import org.edx.mobile.model.db.DownloadEntry;
+import org.edx.mobile.module.db.impl.DatabaseFactory;
+import org.edx.mobile.player.TranscriptManager;
+import org.edx.mobile.util.NetworkUtil;
+import org.edx.mobile.util.VideoUtil;
+
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Locale;
+
+import subtitleFile.Caption;
+import subtitleFile.FormatSRT;
+import subtitleFile.TimedTextObject;
+
+public class CourseUnitYoutubeVideoFragment extends CourseUnitVideoFragment implements YouTubePlayer.OnInitializedListener {
+
+    private static final int SUBTITLES_DISPLAY_DELAY_MS = 50;
+
+    private YouTubePlayer youTubePlayer;
+    private Handler subtitleDisplayHandler = new Handler();
+    private Handler transcriptsHandler = new Handler();
+    private Handler initializeHandler = new Handler();
+    private TimedTextObject subtitlesObj;
+    private LinkedHashMap<String, TimedTextObject> srtList = new LinkedHashMap<>();
+    private YouTubePlayerSupportFragment youTubePlayerFragment;
+    /*
+     * wasHiding is set on true when the app comes to background from foreground
+     * so this allow to play a video when the app comes to foreground from background
+     * fromLandscape allow to know if the user has exited from fullscreen on landscape mode a keep the normal screen size.
+     */
+    private boolean fromLandscape, wasHiding;
```

Comment:
IMO `isFullScreen` and `isInForeground` are more appropriate instated of `fromLandscape` and `wasHiding`. Also update the uses, and doc strings accordingly.

Reply:
We can use the `allowFullScreen` make sure should added proper java doc.